jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -789,7 +789,7 @@ public Resource newResource(URL url) throws IOException
      */
     public Resource newResource(URI uri) throws IOException
     {
-        return ResourceFactory.root().newResource(uri);
+        return ResourceFactory.of(this).newResource(uri);
```

Comment:
Using the root was wrong, but also the new code is wrong.   It should be `ResourceFactory` relative to the resourceBase!